src/view/MapVisualizationView.java

```diff
@@ -1,2 +1,240 @@
+
+    // add markers
+    for (int i = 0; i < vehicleData.getSize(); i++) {
+        MapMarkerDot marker = new MapMarkerDot(
+                trainsLayer,
+                vehicleData.getStringList().get(i),
+                vehicleData.getCoordinateList().get(i),
+                this.defaultStyle
+        );
+        map.addMapMarker(marker);
+    }
+
+    // On initialization, map is focused on the first vehicle within the list
+    // TODO: Implement properly
+    // TODO: What happens if there are no vehicles in the list? Throw an error?
+    Coordinate coordinate = vehicleData.getCoordinateList().get(0);
+    map.setDisplayPosition(coordinate, 16);
+    }
+
+    /**
+     * Invoked when an action occurs.
+     *
+     * @param e the event to be processed
+     */
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        System.out.println("Evoked" + e.getActionCommand());
+    }
+
+    /**
+     * This method gets called when a bound property is changed.
+     *
+     * @param evt A PropertyChangeEvent object describing the event source
+     *            and the property that has changed.
+     */
+    @Override
+    public void propertyChange(PropertyChangeEvent evt) {
+        // TODO: This OK?
+
+        // Ignore any property changes fired from OSM
+        // (adding the map to the screen fires propertyChange)
+        if (!(evt.getSource().toString().startsWith("org.openstreetmap"))) {
+            VisualizeState vehicleData = (VisualizeState) evt.getNewValue();
+            visualizeViewModel.setVisualizationState(vehicleData);
+            new MapVisualizationView(visualizeViewModel).setVisible(true);
+        }
+    }
+
+    class myRenderer extends DefaultListCellRenderer {
+
+        /**
+         * Return a component that has been configured to display the specified
+         * value. That component's <code>paint</code> method is then called to
+         * "render" the cell.  If it is necessary to compute the dimensions
+         * of a list because the list cells do not have a fixed size, this method
+         * is called to generate a component on which <code>getPreferredSize</code>
+         * can be invoked.
+         *
+         * @param list         The JList we're painting.
+         * @param value        The value returned by list.getModel().getElementAt(index).
+         * @param index        The cells index.
+         * @param isSelected   True if the specified cell was selected.
+         * @param cellHasFocus True if the specified cell has the focus.
+         * @return A component whose paint() method will render the specified value.
+         * @see JList
+         * @see ListSelectionModel
+         * @see ListModel
+         */
+        @Override
+        public Component getListCellRendererComponent(JList list, Object value, int index, boolean isSelected, boolean cellHasFocus) {
+            // TODO: Fix handling for bus objects. Here we've type casted the value to display in the list.
```

Comment:
Don't think we will have to worry about bus objects due to the scope of our project

Reply:
Will delete this TODO statement